assets/src/components/selection/ticket-select/index.js

```diff
@@ -0,0 +1,256 @@
+	addAllOptionLabel: __( 'All Tickets', 'event_espresso' ),
+	attributes: {
+		limit: 20,
+		orderBy: 'start_date',
+		order: 'desc',
+		showExpired: false,
+	},
+};
+
+/**
+ * Used to map an orderBy string to the actual value used in a REST query from
+ * the context of an ticket.
+ * @todo this should be moved to a mapper library for various EE Rest Related
+ * things maybe?
+ *
+ * @param {string} orderBy
+ *
+ * @return { string } Returns an actual orderBy string for the REST query for
+ *                      the provided alias
+ */
+const mapOrderBy = ( orderBy ) => {
+	const orderByMap = {
+		start_date: 'TKT_start_date',
+		end_date: 'TKT_end_date',
+	};
+	return isUndefined( orderByMap[ orderBy ] ) ?
+		orderBy :
+		orderByMap[ orderBy ];
+};
+
+/**
+ * Builds where conditions for an tickets endpoint request using provided
+ * information.
+ *
+ * @param {boolean} showExpired 	Whether or not to include expired tickets.
+ * @param {string} month        	Return tickets for the given month. Can be
+ *                              	any in any month format recognized by moment
+ * @param {number} forEventId   	ID for Event to retrieve tickets from
+ * @param {number} forDatetimeId 	ID for Event to retrieve tickets from
+ * @return {string}             	The assembled where conditions.
+ */
+const whereConditions = ( {
+	forEventId = 0,
+	forDatetimeId = 0,
+	showExpired = false,
+	month = 'none',
+} ) => {
+	const where = [];
+	const GREATER_AND_EQUAL = encodeURIComponent( '>=' );
+	const LESS_AND_EQUAL = encodeURIComponent( '<=' );
+
+	if ( ! showExpired ) {
+		where.push(
+			'where[TKT_end_date**expired][]=>' +
+			'&where[TKT_end_date**expired][]=' +
+			nowDateAndTime.local().format()
+		);
+	}
+	if ( month && month !== 'none' ) {
+		where.push(
+			'where[TKT_start_date][]=' + GREATER_AND_EQUAL +
+			'&where[TKT_start_date][]=' +
+			moment().month( month ).startOf( 'month' ).local().format()
+		);
+		where.push(
+			'where[TKT_end_date][]=' + LESS_AND_EQUAL +
+			'&where[TKT_end_date][]=' +
+			moment().month( month ).endOf( 'month' ).local().format()
+		);
+	}
+	forEventId = parseInt( forEventId );
+	if ( forEventId !== 0 && ! isNaN( forEventId ) ) {
+		where.push( 'where[Datetime.Event.EVT_ID]=' + forEventId );
+	}
+	forDatetimeId = parseInt( forDatetimeId );
+	if ( forDatetimeId !== 0 && ! isNaN( forDatetimeId ) ) {
+		where.push( 'where[Datetime.DTT_ID]=' + forDatetimeId );
+	}
+	return where.join( '&' );
+};
+
+/**
+ * The TicketSelect Component wrapped in the `withSelect` higher order
+ * component. This subscribes the TicketSelect component to the state
+ * maintained via the eventespresso/lists store.
+ */
+export default withSelect( ( select, ownProps ) => {
+	const { attributes = TicketSelect.defaultProps.attributes } = ownProps;
+	const {
+		selectedTicketId,
+		forEventId,
+		forDatetimeId,
+		addAllOption,
+		addAllOptionLabel,
+	} = ownProps;
+	attributes.forEventId = forEventId;
+	attributes.forDatetimeId = forDatetimeId;
```

Comment:
because the attributes are being passed to the function that builds the where parameters.
I tried to pass them separately, but it didn't work because that function is... (trying to think of correct term)... _decomposing_... ??? ... the parameters that are passed to within the attributes object.

I tried to search online for another way to do that but don't know any of this new fangled JS lingo

Reply:
Hmm I see.  The `attributes` object prop is used by GB to keep track of things that persist to the db.  Technically this means its only really needed on the top-level component utilizing selects via composition.  So in a way, my event-select example is a bit misleading.  I'm thinking of rectifying that in the work I'm doing in my branch. So, there's a few ways this could be handled here:

- leave it for now as it'll get changed anyways with the modelSelect work.
- don't mutate the attributes directly but instead make `forEventId` and `forDatetimeId` properties of the attributes. 
- just have your `whereConditions` function receive multiple arguments so you simply pass along `forEventId` and `forDatetimeId` as the additional arguments.

I'm fine if you decide to leave as is for now in lieue of pending changes coming down the line.

Reply:
ya I would prefer to just leave things as is for now since this is all going to change.
My primary goal atm is to get the Event Attendees block merged into GBM (Gutenberg/master) which should hopefully be able to continue to use its select components as is, since all of the changes to them will be under the hood.
Once you've solidified the `ModelSelect` component, I will be going back and making the required changes to the child classes that utilize it (such as this one), as well as adding unit tests for those components.